src/service.py

```diff
@@ -39,24 +36,39 @@ def wrapper(self: Any, *args: Tuple[Any], **kwargs: Dict[str, Any]) -> Any:
     return wrapper
 
 
+def get_redfish_conn_params(username, password) -> Dict[str, Any]:
+    """Get redfish connection parameters if redfish is available."""
+    bmc_tools = bmc_hw_verifier()
+    if HWTool.REDFISH not in bmc_tools:
+        logger.warning("Redfish unavailable, disregarding redfish config options...")
+        return {}
+    return {
+        "host": f"https://{get_bmc_address()}",
+        "username": username,
+        "password": password,
+    }
+
+
 class ExporterError(Exception):
     """Custom exception for exporter errors.
 
     This will cause juju to set the charm in ErrorStatus.
     """
 
 
-class ExporterTemplate:
-    """Jinja template helper class for exporter."""
+class BaseExporter(metaclass=ABCMeta):
+    """A class representing the exporter and the metric endpoints."""
+
+    def __init__(self, charm_dir: Path) -> None:
+        """Initialize the class."""
+        self.charm_dir = charm_dir
 
-    def __init__(self, search_path: Path):
-        """Initialize template class."""
-        self.environment = Environment(loader=FileSystemLoader(search_path / "templates"))
-        self.config_template = self.environment.get_template(EXPORTER_CONFIG_TEMPLATE)
-        self.service_template = self.environment.get_template(EXPORTER_SERVICE_TEMPLATE)
+    def __str__(self) -> str:
+        """Get string representation of the exporter name."""
+        return self.__class__.__name__
 
-    def _install(self, path: Path, content: str) -> bool:
-        """Install file."""
+    def _setup_file(self, path: Path, content: str) -> bool:
+        """Setup a file with content."""
```

Comment:
```suggestion
    def _write_to_file(self, path: Path, content: str) -> bool:
        """Write to file with provided content."""
```
This would make the functionality more explicit